.github/workflows/build.yaml

```diff
@@ -10,24 +10,46 @@ on:
 jobs:
   test:
     runs-on: ${{ matrix.os }}
-    name: Build and test on ${{ matrix.os }} ${{ matrix.compiler }}
+    name: Build and test ${{ matrix.build_type }} on ${{ matrix.os }} ${{ matrix.compiler }} ${{ matrix.cmake_generator }}
     strategy:
+      fail-fast: false
       matrix:
         include:
           - os: windows-2019
             cmake_generator: "-G \"Visual Studio 16 2019\" -A x64"
+            build_type: "Release"
+            compiler_flags: "-D_CRT_SECURE_NO_WARNINGS /EHsc"
+          - os: windows-2019
+            cmake_generator: "-G \"Visual Studio 16 2019\" -A x64"
+            build_type: "Debug"
+            compiler_flags: "-D_CRT_SECURE_NO_WARNINGS /EHsc"
           - os: windows-2019
             cmake_generator: "-G \"Visual Studio 16 2019\" -A Win32"
+            build_type: "Release"
+            compiler_flags: "-D_CRT_SECURE_NO_WARNINGS /EHsc"
           - os: ubuntu-latest
             privledges: "sudo"
+            build_type: "Release"
+          - os: ubuntu-latest
+            privledges: "sudo"
+            build_type: "Debug"
+            compiler_flags: "-Wextra -Wall"
```

Comment:
We could add [`-pedantic`](https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#Options-to-Request-or-Suppress-Warnings) if we care about ISO C++ compliance